[python3] Build error on x64-mingw-static

**Host Environment**
- OS: Windows 10
- Compiler: gcc version 12.1.0

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install boost:x64-mingw-static`

**Failure logs**
Installing 131/151 python3:x64-mingw-static...
Building python3[core]:x64-mingw-static...
warning: -- Using community triplet x64-mingw-static. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: C:\vcpkg\triplets\community\x64-mingw-static.cmake
-- Using cached python-cpython-v3.10.5.tar.gz.
-- Extracting source C:/vcpkg/downloads/python-cpython-v3.10.5.tar.gz
-- Applying patch 0001-static-library.patch
-- Applying patch 0002-use-vcpkg-zlib.patch
-- Applying patch 0003-devendor-external-dependencies.patch
-- Applying patch 0004-dont-copy-vcruntime.patch
-- Applying patch 0005-only-build-required-projects.patch
-- Applying patch 0009-python.pc.patch
-- Applying patch 0010-bz2d.patch
-- Using source at C:/vcpkg/buildtrees/python3/src/v3.10.5-bba658437a.clean
-- WARNING: Static builds of Python will not have C extension modules available.
-- Building PCbuild/pcbuild.proj for Release
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: msbuild C:/vcpkg/buildtrees/python3/x64-mingw-static-rel/v3.10.5-bba658437a.clean/PCbuild/pcbuild.proj /p:Configuration=Release /p:IncludeExtensions=false /p:IncludeExternals=false /p:IncludeTests=false /p:ForceImportBeforeCppTargets=C:/vcpkg/buildtrees/python3/src/v3.10.5-bba658437a.clean/PCbuild/python_vcpkg.props /p:IncludeUwp=false /p:_VcpkgPythonLinkage=StaticLibrary /t:Rebuild /p:Platform=x64 /p:PlatformToolset=external /p:VCPkgLocalAppDataDisabled=true /p:UseIntelMKL=No /p:WindowsTargetPlatformVersion= /p:VcpkgTriplet=x64-mingw-static /p:VcpkgInstalledDir=C:/vcpkg/installed /p:VcpkgManifestInstall=false /p:UseMultiToolTask=true /p:MultiProcMaxCount=9 /p:EnforceProcessCountAcrossBuilds=true /m:9 /p:WholeProgramOptimization=false
    Working Directory: C:/vcpkg/buildtrees/python3/x64-mingw-static-rel/v3.10.5-bba658437a.clean
    Error code: 系统找不到指定的文件。(english: system can not found specific file)
    See logs for more information:

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_msbuild.cmake:79 (vcpkg_execute_required_process)
  ports/python3/portfile.cmake:140 (vcpkg_install_msbuild)
  scripts/ports.cmake:146 (include)


error: building python3:x64-mingw-static failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+python3
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[python3]+Build+error
Include '[python3] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-06-17-9268e366206712e38102b28dbd1617697a99ff2e
    vcpkg-scripts version: 98f8d00e8 2022-07-08 (33 hours ago)

You can also use the prefilled template from C:\vcpkg\installed\vcpkg\issue_body.md.

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

--- a/ports/python3/vcpkg-cmake-wrapper.cmake
+++ b/ports/python3/vcpkg-cmake-wrapper.cmake
@@ -89,6 +89,9 @@ if(_PythonFinder_WantLibs)
             find_dependency(Intl)
             if(TARGET @PythonFinder_PREFIX@::Python)
                 get_target_property(_PYTHON_INTERFACE_LIBS @PythonFinder_PREFIX@::Python INTERFACE_LINK_LIBRARIES)
+                if(NOT _PYTHON_INTERFACE_LIBS)
+                    set(_PYTHON_INTERFACE_LIBS "")
+                endif()
                 list(REMOVE_ITEM _PYTHON_INTERFACE_LIBS "-liconv" "-lintl")
                 list(APPEND _PYTHON_INTERFACE_LIBS
                     Iconv::Iconv
@@ -99,6 +102,9 @@ if(_PythonFinder_WantLibs)
             endif()
             if(TARGET @PythonFinder_PREFIX@::Module)
                 get_target_property(_PYTHON_INTERFACE_LIBS @PythonFinder_PREFIX@::Module INTERFACE_LINK_LIBRARIES)
+                if(NOT _PYTHON_INTERFACE_LIBS)
+                    set(_PYTHON_INTERFACE_LIBS "")
+                endif()
                 list(REMOVE_ITEM _PYTHON_INTERFACE_LIBS "-liconv" "-lintl")
                 list(APPEND _PYTHON_INTERFACE_LIBS
                     Iconv::Iconv

--- a/ports/python3/vcpkg.json
+++ b/ports/python3/vcpkg.json
@@ -5,7 +5,7 @@
   "description": "The Python programming language",
   "homepage": "https://github.com/python/cpython",
   "license": "Python-2.0",
-  "supports": "!uwp",
+  "supports": "!uwp & !mingw",
   "dependencies": [
     {
       "name": "bzip2",

--- a/ports/python3/vcpkg.json
+++ b/ports/python3/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "python3",
   "version": "3.10.5",
+  "port-version": 1,
   "description": "The Python programming language",
   "homepage": "https://github.com/python/cpython",
   "license": "Python-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5702,7 +5702,7 @@
     },
     "python3": {
       "baseline": "3.10.5",
-      "port-version": 0
+      "port-version": 1
     },
     "qca": {
       "baseline": "2.3.4",

--- a/versions/p-/python3.json
+++ b/versions/p-/python3.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "c345c4a8ad91847522188517c68e8f83637f4440",
+      "version": "3.10.5",
+      "port-version": 1
+    },
+    {
       "git-tree": "d7c43b7217707bb35a86859d9285496fc2bce8e2",
       "version": "3.10.5",
       "port-version": 0

